extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionManagerBuildTimeConfig.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.narayana.jta.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public final class TransactionManagerBuildTimeConfig {
+    /**
+     * Allow using multiple XA unaware resources in the same transactional demarcation with XA capable resources.
+     * This is UNSAFE and may only be used for compatibility.
+     * <p>
+     * Either use XA for all resources if you want consistency, or split the code into separate
+     * methods with separate transactions.
+     * <p>
+     * Note that using a single XA unaware resource together with XA aware resources, known as
+     * the Last Resource Commit Optimization (LRCO), is different from using multiple XA unaware
+     * resources. Although LRCO allows most transactions to complete normally, some errors can
+     * cause an inconsistent transaction outcome. Using multiple XA unaware resources is not
+     * recommended since the probability of inconsistent outcomes is significantly higher and
+     * much harder to recover from than LRCO. For this reason, use LRCO as a last resort.
+     * <p>
+     * We plan to remove this property in the future so you should plan fixing your application
+     * accordingly.
+     * If you think your use case of this feature is valid and this option should be kept around,
+     * open an issue in our tracker explaining why.
+     *
+     * @deprecated This property is planned for removal in a future version.
+     */
+    @Deprecated(forRemoval = true)
```

Comment:
Does this mean this property is only introduced in `3.8.x`, and in no other version? If not: when will it be removed?

Reply:
The original patch Guillaume proposed was more explicit on it being removed in a specific version, but I suggested to keep our options open, so now it's a bit more vague.
I believe we'll want to remove it hence the stern warning, but when exactly can depend on feedback and possibly on reasonable use cases being presented.

Reply:
Why not add the deprecation when we know the *when*-part? As of now, Everyone will have the warning. When we finally know when to remove it, nobody will "see" the warning anymore.

Reply:
Because we don't know what the valid use cases for this are exactly, so we need feedback before we can provide a migration path. See:

```
     * If you think your use case of this feature is valid and this option should be kept around,
     * open an issue in our tracker explaining why.
```

The idea being that people seeing this deprecation will either not use the property, or use it and provide the feedback as requested, knowing that they'll eventually be in trouble. If they don't reach out, it's on them.